internal/server/events.go

```diff
@@ -127,6 +128,31 @@ func (s *Server) eventHandler() http.HandlerFunc {
 	}
 }
 
+func (s *Server) createEventHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var e store.Event
+		if err := json.NewDecoder(r.Body).Decode(&e); err != nil {
+			ihttp.Error(w, http.StatusUnprocessableEntity)
+			return
+		}
+
+		e.ManuallyAdded = true
+
+		// this is redundant since the route should be admin-protected anyways
+		if !getRoles(r).IsAdmin {
+			ihttp.Error(w, http.StatusForbidden)
```

Comment:
What do you think of logging an error here, so that if for some reason a non-admin was able to get to this point it'd show up in the error log and we could figure out what broke?

Reply:
Sounds like a good idea to me :+1: